### PR TITLE
chore: rewrite auction.js unit test

### PR DESCRIPTION
**Pull request by imkrish**
mainly test
- bet method
- withdraw method

### CODE REVIEW

File: test/auction.js [hunk 1]
```diff
@@ -34,187 +37,308 @@ contract(
...
-
-      const currentAuctionId = await dailyauction.currentAuctionId();
-
-      // Change node time and swap
-      await helper.advanceTimeAndBlock(DAY * 1);
-
-      await dailyauction.withdraw(currentAuctionId, {
-        from: account1,
+    describe("withdraw", () => {
+      describe("failure cases", () => {
```
Comment (Vicarious1) [inline test/auction.js:195]: I've never used Mocha before and didn't realize you could nest `describe` like this. I like the resulting output.

I think `describe` and `contract` are very similar https://www.trufflesuite.com/docs/truffle/testing/writing-tests-in-javascript

It looks like the difference is that `contract` re-deploys smart contracts to the test blockchain to provide a clean environment for tests. It doesn't look like that's needed here, but I'm curious is there a specific reason you used `describe` over `contract`?

**Review by Vicarious1 — APPROVED** [review]
The test cases look good, the logic is sound and matches the whitepaper and how the site functioned on testnet. Thanks!

File: test/auction.js (hunk 1 above)
Comment (imkrish) [reply test/auction.js:195]: @Vicarious1 
I use nested "describe" here as when we run the test it is easier to read

Contract: Auction
    bet
      ✓ should update the contract state correctly (537ms)
    withdraw
      failure cases
        ✓ should fail if the user withdraws before betting (118ms)
        ✓ should fail if the auction is still active (122ms)
      successful cases
        ✓ should success and update the contract state correctly (without ref) (694ms)
        ✓ should success and update the contract state correctly (with ref) (780ms)

We use contract('xxx') to say that this test is for contract 'xxx'
And use describe grouping the test result

Comment (imkrish) [reply test/auction.js:195]: Thanks for the review mate

**Pull request merged**